### DISCUSSION TITLE
Typescript error with Select compoent

### DISCUSSION THREAD

**Discussion by jkiss** [heroui-inc/heroui#2477] [Q&A]
```
<Select
    onSelectionChange={(keys) => {
        keys.size > 0 && setOutputLang(keys)    // Here will trigger a ts error like below
    }}
>
</Select>
```

![image](https://github.com/nextui-org/nextui/assets/3231326/a792dd51-8dd1-457e-9f64-26b53feead95)

Is it a BUG?

**Comment by alphaxek** [comment]
Hi @jkiss, can you please share the code snippet to reproduce the issue on my side, thanks

**Reply by jkiss** [reply]
[reproduce on codesandbox](https://codesandbox.io/p/devbox/brave-river-9nynpg?file=%2Fapp%2Fpage.tsx%3A25%2C48)

**Comment by wingkwong** [comment]
@jkiss This is just a typing error. In fact the message already shows what is wrong.

Since keys has the type `Selection` which is `'all' | Set<Key>`, you can't just call `.size` since this is for `Set` only. Therefore, in order to get rid of the typing error, you may try the following

```ts
const size = typeof keys !== "string" ? keys.size : data.length; // update data.length yourself for your case
if (size > 0) setOutputLang(keys)
```

**Reply by jkiss** [reply]
ok, your code works for me, but why `keys` has a 'all' type value? It only should be Set<Key>, right? really confuse about this 😢

**Reply by wingkwong** [reply]
@jkiss The type `Selection` is actually defined in `@react-types/shared`. The reason behind is to cover the select-all event, which could be a checkbox to select all items in the collection or shortcut like `cmd + A`. If `onSelectionChange` is set with `all`, then `selectedKeys` and `defaultSelectedKeys` will be set to `all` programmatically.

**Reply by jkiss** [reply]
it's a simple way to add a check mark behind every selected item, BUT 'all' just a string, it's NOT selected keys Set, right? It also make selectedKeys state hard to use, users must get all keys like: `const allKeys = selectedKeysState === 'all' && [key1, key2, ...]` 😕
